ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/common/logging/NamedLogger.scala

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.common.logging
+
+import org.slf4j.{Logger, LoggerFactory}
+
+/**
+  * NamedLogger augments a regular class-based slf4j logger with one annotated with a "name" where the name provides
+  * human readable context identifying a class instance, e.g. which participant in a set of participants has logged
+  * a particular message.
+  *
+  * The name can be constructed in a nested, left-to-right append manner.
+  */
+trait NamedLogger {
```

Comment:
Not really a logger though, but rather a factory of loggers.
```suggestion
trait NamedLoggerFactory {
```
I would also suggest being explicit about this when naming the instances around the codebase (i.e. `loggerFactory` rather than `namedLogger`).

Reply:
👍